spyql/processor.py

```diff
@@ -129,6 +129,27 @@ def handle_1st_data_row(self, row):
         # dict of {col1: value1, ...}
         self.translations["row"] = f"NullSafeDict(zip(_names, {cols_expr}))"
 
+    def expand_col_names(self, all_col_names, name):
+        """
+        Expands column wildcard operator and modifiers
+        """
+        except_pattern = re.compile(r"\*\s*EXCEPT\s*\((.*)\)", re.IGNORECASE)
+        except_match = re.search(except_pattern, name)
+
+        if name == "*":
+            return all_col_names
+        elif except_match:
+            except_col_names = [
+                col_name.strip() for col_name in except_match.group(1).split(",")
+            ]
```

Comment:
I would move this to the parsing stage, where you already have the regex pattern defined. e.g. `expr` could be list/set of strings.
In general, I think we should avoid doing parsing in the processor.

Reply:
We would need to think in a way of signalling that an expression is an EXCEPT pattern. We could either use an expression name like `"*EXCEPT"` or `"*-"`, or we could use an extra field/flag... e.g.
```
{
    "name": "*EXCEPT"
    "expr": ["a column to exclude", "another1"]
}
```

Reply:
I thought about this but it felt strange to split the statement between the name and expr. They kind of lose their semantic meaning but it does make the code simpler.